client/hijack.go

```diff
@@ -59,10 +64,31 @@ func fallbackDial(proto, addr string, tlsConfig *tls.Config) (net.Conn, error) {
 	return net.Dial(proto, addr)
 }
 
-func (cli *Client) setupHijackConn(ctx context.Context, req *http.Request, proto string) (net.Conn, string, error) {
+func (cli *Client) setupHijackConn(ctx context.Context, req *http.Request, proto string) (_ net.Conn, _ string, retErr error) {
 	req.Header.Set("Connection", "Upgrade")
 	req.Header.Set("Upgrade", proto)
 
+	// We aren't using the configured RoundTripper here so manually inject the trace context
+	tp := cli.tp
+	if tp == nil {
+		if span := trace.SpanFromContext(ctx); span.SpanContext().IsValid() {
+			tp = span.TracerProvider()
+		} else {
+			tp = otel.GetTracerProvider()
+		}
+	}
+
+	ctx, span := tp.Tracer("").Start(ctx, req.Method+" "+req.URL.Path)
```

Comment:
I could add a comment, though I suspect after this we may want to add more custom spans where we'd be doing the same thing.
To me the documentation is in the interface.

Reply:
Oh! question on the same line; as we're shadowing `ctx`  here, would it be clearer to;

```suggestion
	var span trace.Span
	ctx, span = tp.Tracer("").Start(ctx, req.Method+" "+req.URL.Path)
```

Reply:
Why?

Reply:
Ah, sorry, yeah currently it doesn't add much. I ended up in a rebase / branch where this line ended up in a "conditional" (if), in which case the "shadowing" became an actual "shadow". Hence my thinking if it would be better to be defensive here. But agreed, that it's not super-important.